pruntime/enclave/src/lib.rs

```diff
@@ -1092,6 +1106,41 @@ fn sync_block(input: SyncBlockReq) -> Result<Value, Value> {
     }))
 }
 
+fn sync_events(input: SyncEventsReq) -> Result<Value, Value> {
+	let mut state = STATE.lock().unwrap();
+
+	let mut root: [u8; 32] = Default::default();
+	root.copy_from_slice(&input.root);
+	let state_root = Hash::from(root);
+	let validation = state.light_client.validate_events_proof(&state_root, input.proof, input.events.clone(), input.key);
```

Comment:
If we combine the block and event syncing together, the state root is right the same value in the block header. By this way, we can make sure the events actually come from a certain block, because:
1. The block is validated
2. The events are validated
3. The events have a valid merkle proof that commits to the block